HACKING.md

```diff
@@ -2,34 +2,23 @@
 
 # Hacking on the MCD
 
-1. Create a cluster (e.g. using [libvirt](https://github.com/openshift/installer/blob/master/Documentation/dev/libvirt-howto.md))
-2. Delete the old node agent daemonset:
-
-    ```sh
-    oc delete ds tectonic-node-agent --namespace tectonic-system
-    ```
-
-    This will be fixed in the installer eventually.
-
-3. Run:
-
-    ```sh
-    sed s/node-configuration.v1.coreos.com/machineconfiguration.openshift.io/g /opt/tectonic/node-annotations.json >/etc/machine-config-daemon/node-annotations.json
-    ```
-
-    The install process will take care of this eventually.
-
-4. Create a namespace, e.g. `oc new-project mco`
-5. Create items from `manifests/` in a sensible order, e.g.
-
-    ```sh
-    for f in manifests/*.crd.yaml; do oc create -f $f; done
-    for f in manifests/*.machineconfigpool.yaml; do oc create -f $f; done
-    oc create -f manifests/machineconfigdaemon/clusterrole.yaml
-    oc create -f manifests/machineconfigdaemon/sa.yaml
-    oc create -f manifests/machineconfigdaemon/scc.yaml
-    oc create -f manifests/machineconfigdaemon/clusterrolebinding.yaml
-    oc create -f manifests/machineconfigdaemon/daemonset.yaml
-    ```
-
-    (Though note you'll have to sub in `{{.TargetNamespace}}` with `mco`).
+These instructions were tested with installer version https://github.com/openshift/installer/commit/d0b3f913981ef79fd58f089909d7ef1918aa3894 and RHCOS version 4.0.5836.
+
+1. Create a cluster (e.g. using [libvirt](https://github.com/openshift/installer/blob/d0b3f913981ef79fd58f089909d7ef1918aa3894/Documentation/dev/libvirt-howto.md))
+2. Build a container image for the MCD and push it to a registry somewhere, e.g.
+
+   ```sh
+   # this takes care of building the binary for you as well
+   docker build -t sdemos/mcd:<tag> -f Dockerfile .
```

Comment:
`WHAT=machine-config-daemon ./hack/build-image.sh`

Reply:
the problem with using that script is it doesn't provide a way to change the image repository you are tagging for

Reply:
I use podman to push images:
```sh
#!/usr/bin/env bash

set -eu

# Print errors to stderr
function print_error {
        echo "ERROR: $1" >&2
}

function print_info {
        echo "INFO: $1" >&2
}

# Warn when unprivileged
if [ `id --user` -ne 0 ]; then
        print_error "Note: Running unprivileged may fail due to permissions"
fi

# Record all images files
ALL_IMAGES=Dockerfile.*

# To push will hold a list of all image images to push
TOPUSH=""

REPO=${REPO:-"openshift"}

# WHAT should be defined. If not, give a list and exit
if [ -z ${WHAT+a} ]; then
        print_error "WHAT must be set to one of the following:"
        print_error "- all"
        for x in $ALL_IMAGES ; do
                print_error "- ${x#Dockerfile.}"
        done

        exit 1
fi


# If all is the WHAT target then set TOPUSH to all the images found
if [ ${WHAT} == "all" ]; then
        TOPUSH=$ALL_IMAGES
        print_info "Pushing all images"
else
        # Otherwise WHAT should be valid at this point
        TOPUSH="${WHAT}"
fi


if [ -z ${VERSION+a} ]; then
        print_info "Using version from git..."
        VERSION=$(git describe --abbrev=8 --dirty --always)
fi

# Push all images requested
for IMAGE_TO_PUSH in $TOPUSH; do
        NAME="${IMAGE_TO_PUSH#Dockerfile.}"
        set -x
        podman push "${NAME}:${VERSION}" "${REPO}/origin-${NAME}:${VERSION}"
        podman push "${NAME}:${VERSION}" "${REPO}/origin-${NAME}:latest"
        set +x
done

```

Reply:
@abhinavdahiya that looks like a good enhancement to the image build script. Do you mind PRing it (or point us to it if it's already in a PR)?

Reply:
Once the above lands I second updating this doc. In the meantime I think this looks good.

Reply:
ooohhh I didn't realize podman could a built image to an arbitrary remote repository, that's way better.

time to install podman...